listenbrainz/listenstore/influx_listenstore.py

```diff
@@ -348,13 +348,14 @@ def fetch_listens_from_storage(self, user_name, from_ts, to_ts, limit, order):
         return listens
 
 
-    def dump_user(self, username, fileobj, dump_time):
+    def dump_user(self, username, fileobj, dump_time, spark_format=False):
         """ Dump specified user's listens into specified file object.
 
         Args:
             username (str): the MusicBrainz ID of the user whose listens are to be dumped
             fileobj (file): the file into which listens should be written
             dump_time (datetime): the time at which the specific data dump was initiated
+            spark_format (bool): dump files in Apache Spark friendly format if True, else full dumps
```

Comment:
would it be possible to do these dumps without making the dump methods have to understand an implementation detail (spark)? e.g., could you have a method which converts the result of `to_api()` to the spark format, and you call that in the spark dumper?

Reply:
In general I agree with this, but this may end up being premature optimization. Part of it is that we simply do not yet know what what we really need and how we need it, in order to feed Spark. At this point I would be happy to enter a ticket for us to revisit this and proceed for now.